### DISCUSSION TITLE
Add CDN option to use custom CDN such as BunnyCDN, Fastly, etc. for improved performance.

### DISCUSSION THREAD

**Discussion by iloveoss** [GetPublii/Publii#1786] [Ideas]
Need a field possibly under performance section that allows users to enter custom CDN url; for e.g. https://cdn.website.com.

Then this CDN url will be applied to all the static assets. Its more like regex search and replace. You can even add option to whitelist and blacklist extensions .png, .svg, .etc.

**Comment by dziudek** [comment]
Hi,

Such replacements can be done with Smart Replacer plugin: https://marketplace.getpublii.com/plugins/smart-replacer/ 

Of course you have to manage the upload of your assets to your CDN location. 

At this moment we have no plans to add such functionality to the core as it will complicate the sync process, and the easiest way is to use Github Pages and actions which will upload assets to CDN automatically.

**Reply by vivek-kumar-poddar** [reply]
This is not how CDNs work. You don't "upload" assets to CDN server. CDN servers automatically download (better said cache) the requested static file (styles, js, videos, images) to their server. All you do is, add a subdomain like cdn or assets to the domain name (CNAME). So if its domainname.com/assets/js/main.js — it will become — cdn.domainname.com/assets/js/main.js

That's all.

But given this app (Publii) renders the files for local preview and then upload the files to the server, the cdn subdomain should be applied only when uploaded to the server. The assets (images, js, css, etc) will remain in their respective folder (nothing needs to be changed).

Now when the request is made to the server, for example:

domain.com -> which will load -> cdn.domainname.com/assets/images/demo.png (as in the html code) -> the cdn server will make request to -> domainname.com/assets/images/demo.png to cache the file. Once cached, now all the request to load domainname.com/assets/images/demo.png will be served from the CDN cdn.domainname.com/assets/images/demo.png

**Reply by dziudek** [reply]
@vivek-kumar-poddar - well, in fact there are two types of CDN - push CDN and pull CDN. And in the past we have some questions about support for push CDNs. That's why I have mentioned that we have no plans to handle such upload directly in Publii :) But yes - in pull CDN the only thing to do is replacing URLs of assets and then my solution with smart replacer plugin is enough :)

**Comment by candidexmedia** [comment]
See: https://github.com/GetPublii/Publii/discussions/2107
